SH4RE/SH4RE/Components/Stars.swift

```diff
@@ -8,27 +8,39 @@
 import SwiftUI
 
 struct StarsView: View {
-    private var numberOfStars: Float
+    private var numberOfStars: Int
     private var hasHalfStar: Bool
+    private var remainingStars: Int
     
     init(numberOfStars: Float) {
-        self.numberOfStars = numberOfStars
+        self.numberOfStars = Int(numberOfStars)
         self.hasHalfStar = round(numberOfStars.truncatingRemainder(dividingBy: 1)) == 1 ? true : false
+        
+        self.remainingStars = self.hasHalfStar ? 5 - self.numberOfStars - 1 : 5 - self.numberOfStars
     }
     
     var body: some View {
-        ForEach(0 ..< Int(numberOfStars), id: \.self) { idx in
-            Label("star\(idx)", systemImage: "star.fill")
-                .labelStyle(.iconOnly)
-                .foregroundColor(Color("Yellow"))
-                .font(.system(size: 12))
-        }
-        
-        if (self.hasHalfStar) {
-            Label("half-star", systemImage: "star.fill.left")
-                .labelStyle(.iconOnly)
-                .foregroundColor(Color(UIColor(named: "Yellow")!))
-                .font(.system(size: 12))
+        HStack {
+            ForEach(0 ..< numberOfStars, id: \.self) { idx in
+                Label("star\(idx)", systemImage: "star.fill")
+                    .labelStyle(.iconOnly)
+                    .foregroundColor(Color("Yellow"))
+                    .font(.system(size: 12))
+            }
+            
+            if (self.hasHalfStar) {
+                Label("half-star", systemImage: "star.fill.left")
+                    .labelStyle(.iconOnly)
+                    .foregroundColor(Color(UIColor(named: "Yellow")!))
```

Comment:
nbd, ui color here but not in the other two foreground

Reply:
good catch